### PR TITLE
feat(table): support native table selectors

### LINKED ISSUE #5888: [Table] Add a colspan input for columns
Will mimic the native `<table>` column `colspan` property and allow column cells to stretch across multiple columns

### LINKED ISSUE #6058: md-table: column misalignment in dense tables
#### Bug, feature request, or proposal:
Bug. We are currently using md-table in an application that displays complex tables with a lot of columns containing financial data. The tables oftentimes become rather dense and there is not a lot of room left for each column. If a cell's content is too long and cannot be wrapped, the cell grows in size shifting the row's remaining cells to the right. In our case the problem gets even worse because we need to indent the content of the first cell (using padding-left) in order to visualize hierarchical information.

#### What is the expected behavior?
md-table should behave like an HTML table and ensure all cells in the same column have the same width.

#### What are the steps to reproduce?
I have created a plunker that shows the problem:
https://plnkr.co/edit/8FP4LQFkOnN1EpFbIbeq?p=preview

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
material@2.0.0-beta.8, angular 4.3.1

Sorry if I should have overlooked any issue already describing this problem and thanks for all your work on this great library! :)

### PR DESCRIPTION
**Pull request by andrewseguin**
**Summary:**
Add native table selectors to the `CdkTable` (e.g. `<table cdk-table>`) and refocuses the docs on this approach as the primary way of using the table components

Fixes #6058, #7964, #9027, #8680
Closes #5888, #5866

**Full Description:**
Add the ability to render the table using the native HTML table element as opposed to using the flex-based `<mat-table>` element. This quickly enables key features such as auto column resizing, col/row span, easy copy/paste (cells map into spreadsheets easily). Also, any native table can become a material table by simply applying the `mat-table` class (+ classes to rows/cells).

Benefits to using native table:
- More intuitive approach - use table elements
- Auto-resizing columns
- Colspan and rowspan
- Ability to copy/paste cells into spreadsheets
- Better sticky positioning support
- Increased a11y support (aria + `<table>`)
- `CdkTable` supports a table appearance out of the box (no need for additional styling)
- Header/footer sections show up on all pages if printed
- Better horizontal scrolling (flex doesn't stretch the row styling easily)

### CODE REVIEW

File: src/cdk/table/table.ts [hunk 1]
```diff
@@ -212,14 +212,18 @@ export class CdkTable<T> implements CollectionViewer, OnInit, AfterContentChecke
...
+              private readonly _elementRef: ElementRef,
               @Attribute('role') role: string) {
     if (!role) {
-      elementRef.nativeElement.setAttribute('role', 'grid');
+      this._elementRef.nativeElement.setAttribute('role', 'grid');
     }
   }
 
   ngOnInit() {
+    if (this._elementRef.nativeElement.tagName === 'TABLE') {
```
Comment (jelbourn) [inline src/cdk/table/table.ts]: Prefer `nodeName` to `tagName`

File: src/cdk/table/table.ts [hunk 2]
```diff
@@ -558,4 +562,16 @@ export class CdkTable<T> implements CollectionViewer, OnInit, AfterContentChecke
       return column.cell;
     });
   }
+
+  /** Adds native table sections (e.g. tbody) and moves the row placeholders into them. */
+  private _applyNativeTableSections() {
+    const tHeadEl = document.createElement('thead');
+    const tBodyEl = document.createElement('tbody');
```
Comment (jelbourn) [inline src/cdk/table/table.ts]: Just `thead` and `tbody` since that's the tag name; it's weird as camelCase

File: src/lib/table/table.md [hunk 3]
```diff
@@ -278,3 +278,50 @@ Table's default role is `grid`, and it can be changed to `treegrid` through `rol
 
 `mat-table` does not manage any focus/keyboard interaction on its own. Users can add desired 
 focus/keyboard interactions in their application.
+
+### Using flex layout
```
Comment (jelbourn) [inline src/lib/table/table.md]: Reword this to be something like `` Tables with `display: flex `` so that it's not confused with `@angular/flex-layout`

File: src/lib/table/table.md [hunk 4]
```diff
@@ -278,3 +278,50 @@ Table's default role is `grid`, and it can be changed to `treegrid` through `rol
...
+
+The `MatTable` does not require that you use a native HTML table. Instead, you can use an
+alternative approach that uses flex-layout for the table's styles.
```
Comment (jelbourn) [inline src/lib/table/table.md]: `display: flex`

**Comment by andrewseguin** [comment]
Thanks - updated with the comments applied

**Comment by jainankit102** [comment]
Can you please give us a stackblitz example for this. Having difficulty to implement native table as a material table.

Sample code is also more then enough.

Thanks in advance.

**Comment by CDDelta** [comment]
@jainankit102 check material.angular.io. it's all there.

**Comment by jainankit102** [comment]
@CDDelta thanks for that but I already checked in material.angular.io. didn't found any example regarding material table with rowspan and colspan.

Thanks

**Comment by CDDelta** [comment]
@jainankit102 sorry it was early in the morning, I didn't notice you were looking for rowspan examples. I recommend you open a new issue about this, you would be more likely to get responses.

**Comment by jainankit102** [comment]
Thanks @CDDelta for the help. I have raised the ticket for that i.e. #11551
